File: js/anagram2.js
exports.anagramsFor = function(word, listOfWords) {
    const result = [];
    string1 = word.toLowerCase();
    let arr1 = string1.split(' ').join('').split('').sort();
    for (let i = 0; i < listOfWords.length; i++){
        let string2 = listOfWords[i].toLowerCase();
        let arr2 = string2.split(' ').join('').split('').sort(); 
        if ((arr1.length !== arr2.length)){
            continue;
        }
        for (let j = 0; j < arr1.length; j++){	
            if (arr1[j] !== arr2[j]){	
              continue;	
            }	
        }
        result.push(listOfWords[i]);
    }
    return result;
};
console.log(exports.anagramsFor('threads', ["threads", "trashed", "hardest", "hatreds", "hounds"]))

File: python/anagram2.py
def anagrams_for(word, list_of_words):
	result = []
	string1 = word.upper()
	arr1 = list(''.join(string1.split(' ')))
	arr1.sort()
	for i in range(0, len(list_of_words)):
		string2 = list_of_words[i].lower()
		arr2 = list(''.join(string2.split(' ')))
		arr2.sort()
		if (len(arr1) != len(arr2)):
			continue
		for j in range(0, len(arr1)):
			if (arr1[j] != arr2[j]):
				continue
		result.append(list_of_words[i])
	return(result)